index.js

```diff
@@ -84,18 +84,88 @@ module.exports = {
     }
   },
 
+  _shouldIncludeHelpers() {
+    let appOptions = (this.app && this.app.options) || {};
```

Comment:
I'm concerned with this...

`this.app` will only be present for top level addons (direct dependencies of the project), but including or not include helper should be a global thing. As it stands, we will only ever return `true` for direct project deps and any nested addons would continue to embed their helpers...